dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -0,0 +1,301 @@
+import static org.hisp.dhis.webapi.controller.event.mapper.OrderParamsHelper.toOrderParams;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import lombok.RequiredArgsConstructor;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hisp.dhis.common.DimensionalItemObject;
+import org.hisp.dhis.common.OrganisationUnitSelectionMode;
+import org.hisp.dhis.common.QueryFilter;
+import org.hisp.dhis.common.QueryItem;
+import org.hisp.dhis.feedback.BadRequestException;
+import org.hisp.dhis.feedback.ForbiddenException;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.organisationunit.OrganisationUnitService;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramService;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.trackedentity.TrackedEntityAttributeService;
+import org.hisp.dhis.trackedentity.TrackedEntityQueryParams;
+import org.hisp.dhis.trackedentity.TrackedEntityType;
+import org.hisp.dhis.trackedentity.TrackedEntityTypeService;
+import org.hisp.dhis.trackedentity.TrackerAccessManager;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.webapi.controller.event.mapper.OrderParam;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Maps {@link TrackedEntityOperationParams} to {@link TrackedEntityQueryParams}
+ * which is used to fetch tracked entities from the DB.
+ */
+@Component
+@RequiredArgsConstructor
+class TrackedEntityOperationParamsMapper
+{
+    @Nonnull
+    private final OrganisationUnitService organisationUnitService;
+
+    @Nonnull
+    private final ProgramService programService;
+
+    @Nonnull
+    private final TrackedEntityTypeService trackedEntityTypeService;
+
+    @Nonnull
+    private final TrackedEntityAttributeService attributeService;
+
+    @Nonnull
+    private final TrackerAccessManager trackerAccessManager;
+
+    @Transactional( readOnly = true )
+    public TrackedEntityQueryParams map( TrackedEntityOperationParams operationParams )
+        throws BadRequestException,
+        ForbiddenException
+    {
+        Program program = validateProgram( operationParams.getProgramUid() );
+        ProgramStage programStage = validateProgramStage( operationParams, program );
+        TrackedEntityType trackedEntityType = validateTrackedEntityType( operationParams.getTrackedEntityTypeUid() );
+
+        User user = operationParams.getUser();
+        Set<OrganisationUnit> orgUnits = validateOrgUnits( user, operationParams.getOrganisationUnits(), program );
+        if ( operationParams.getOrganisationUnitMode() == OrganisationUnitSelectionMode.CAPTURE && user != null )
```

Comment:
could this not be done in the method `validateOrgUnits` or in a new one?
also, ACCESSIBLE is very similar to CAPTURE mode, what happens in that case?

Reply:
Moved inside the `validateOrgUnits` method.
I guess this is not working correctly 100%, anyway, I will leave any functional change to this part for the ACL epic.